### PR TITLE
test: improve coverage on 5 files below 80%

### PR DESCRIPTION
**Pull request by hugalafutro**
## Coverage Improvements

| File | Before | After |
|------|--------|-------|
| Chat.tsx | 68.42% | **89.74%** |
| Dashboard.tsx | 66.67% | **100%** |
| AppLogs.tsx | 76.32% | **84.26%** |
| VirtualLogTable.tsx | 79.63% | improved |
| version.go | 79.55% | **90.8%** |

## New Test Files

- `Chat.conversation.test.tsx` (27 tests) — conversation stats bar, attachment preview/handling, action buttons
- `Dashboard.modals.test.tsx` (19 tests) — modal close handlers, metric toggles, per-section settings, usage panels
- `AppLogs.pagination.test.tsx` (12 tests) — paginate mode rendering, filtering, sorting, view mode toggle
- `VirtualLogTable.scroll.test.tsx` (25 tests) — scroll position adjustment, ResizeObserver sync, edge cases

## Bug Fixes

- Fix 3 failing tests in `VirtualAppLogTable.test.tsx` — wrong CSS class (`text-(--accent)` → `text-blue-400`)
- Fix 4 failing tests in `AppLogs.test.tsx` — wrong selector (`/Toggle live/` → `/^Live$/`)

## Other

- Exclude frontend tests from Codacy `metric` (complexity) engine, matching existing `duplication` exclusions
- Reviewed by oracle — addressed all actionable findings (removed dead code, replaced `getByTestId` with `getByRole`)

### CODE REVIEW

File: web/src/pages/AppLogs.tsx [hunk 1]
```diff
@@ -458,114 +345,36 @@ export function AppLogs() {
...
+									pickerMonth={pickerMonth}
+									pendingFrom={pendingFrom}
+									pendingTo={pendingTo}
+									onCalendarSelect={(dateStr) => handleCalendarSelect(dateStr)}
+									onApply={applyDateFilter}
+									onClear={clearDateFilter}
+									onClose={closeDatePicker}
+									anchor="right"
+								/>
+							)}
```
[the remark this answers is not in the document]
Comment (hugalafutro) [reply web/src/pages/AppLogs.tsx]: Fixed in `8a0de62`. Restored `<div className="relative" ref={datePickerRef}>` wrapper around DateFilterButton and DateRangePickerPopover in both AppLogs.tsx and Logs.tsx. Also destructured `datePickerRef` from `useDateRangePicker` in both files.

File: web/src/utils/logBadgeUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+		const result = getSourceBadgeClasses("unknown");
+		expect(result).toBe("bg-gray-800/30 text-gray-400");
+	});
+});
+
+describe("formatTimestamp", () => {
+	it("formats valid ISO date string", () => {
+		const result = formatTimestamp("2024-01-15T10:30:45Z");
+		expect(result).toMatch(/^\d{1,2}\/\d{2}\/\d{4},\s\d{2}:\d{2}:\d{2}$/);
+	});
```
[the remark this answers is not in the document]
Comment (hugalafutro) [reply web/src/utils/logBadgeUtils.test.ts:187]: Fixed in `8a0de62`. Changed `toLocaleString(undefined, ...)` to `toLocaleString("en-US", ...)` in the source, and updated the test to construct the expected string using the same `toLocaleString("en-US", ...)` call instead of a locale-sensitive regex.
